grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -284,7 +288,11 @@ class PrepaymentInvoiceService {
         }
 
         items.each { Map item ->
-            updateInvoiceItemQuantity(item.id, item.quantity)
+            if (item.quantity) {
+                updateInvoiceItemQuantity(item.id, item.quantity)
+            } else if (item.unitPrice) {
+                updateInvoiceItemUnitPrice(item.id, item.unitPrice)
+            }
```

Comment:
btw, I am concerned about not allowing updating quantity and unit price at the same time. Are we sure that we never ever would like to do that?

Reply:
And I am concerned that you'd like to do that :sweat_smile: 

1. On invoice items from order items or shipment items, the unit price should always be based on the unit price that is on the order item (or prepayment invoice item for inverses), we should not allow changing that.
2. On invoice items from order adjustments the quantity is 1 or 0, we don't want to change from one or another.

Taking these two assumptions into account - we don't want to change both fields at the same time for this type of invoice.